aws/ec2.go

```diff
@@ -110,3 +113,394 @@ func nukeAllEc2Instances(session *session.Session, instanceIds []*string) error
 	logging.Logger.Infof("[OK] %d instance(s) terminated in %s", len(instanceIds), *session.Config.Region)
 	return nil
 }
+
+func GetEc2ServiceClient(region string) ec2iface.EC2API {
+	return ec2.New(newSession(region))
+}
+
+type Vpc struct {
+	Region string
+	VpcId  string
+	svc    ec2iface.EC2API
+}
+
+func NewVpcPerRegion(regions []string) []Vpc {
+	var vpcs []Vpc
+	for _, region := range regions {
+		vpc := Vpc{
+			svc:    GetEc2ServiceClient(region),
+			Region: region,
+		}
+		vpcs = append(vpcs, vpc)
+	}
+	return vpcs
+}
+
+func GetDefaultVpcId(vpc Vpc) (string, error) {
+	input := &ec2.DescribeVpcsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   awsgo.String("isDefault"),
+				Values: []*string{awsgo.String("true")},
+			},
+		},
+	}
+	vpcs, err := vpc.svc.DescribeVpcs(input)
+	if err != nil {
+		logging.Logger.Errorf("[Failed] %s", err)
+		return "", errors.WithStackTrace(err)
+	}
+	if len(vpcs.Vpcs) == 1 {
+		return awsgo.StringValue(vpcs.Vpcs[0].VpcId), nil
+	} else if len(vpcs.Vpcs) > 1 {
+		// More than one VPC in a region should never happen
+		err = fmt.Errorf("Impossible - more than one VPC found in region %s", vpc.Region)
+		return "", errors.WithStackTrace(err)
+	}
+	// No default VPC
+	return "", nil
+}
+
+func GetDefaultVpcs(vpcs []Vpc) ([]Vpc, error) {
+	for i := 0; i < len(vpcs); i++ {
+		vpcId, err := GetDefaultVpcId(vpcs[i])
+		if err != nil {
+			return []Vpc{Vpc{}}, errors.WithStackTrace(err)
+		}
+		if vpcId != "" {
+			vpcs[i].VpcId = vpcId
+		} else {
+			// Ignore regions that don't have a default VPC
+			vpcs = append(vpcs[:i], vpcs[i+1:]...)
+			i--
+		}
+	}
+	return vpcs, nil
+}
+
+func (v Vpc) nukeInternetGateway() error {
+	input := &ec2.DescribeInternetGatewaysInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   awsgo.String("attachment.vpc-id"),
+				Values: []*string{awsgo.String(v.VpcId)},
+			},
+		},
+	}
+	igw, err := v.svc.DescribeInternetGateways(input)
+	if err != nil {
+		return errors.WithStackTrace(err)
+	}
+
+	if len(igw.InternetGateways) == 1 {
+		logging.Logger.Infof("...detaching Internet Gateway %s", awsgo.StringValue(igw.InternetGateways[0].InternetGatewayId))
+		_, err := v.svc.DetachInternetGateway(
+			&ec2.DetachInternetGatewayInput{
+				InternetGatewayId: igw.InternetGateways[0].InternetGatewayId,
+				VpcId:             awsgo.String(v.VpcId),
+			},
+		)
+		if err != nil {
+			return errors.WithStackTrace(err)
+		}
+
+		logging.Logger.Infof("...deleting Internet Gateway %s", awsgo.StringValue(igw.InternetGateways[0].InternetGatewayId))
+		_, err = v.svc.DeleteInternetGateway(
+			&ec2.DeleteInternetGatewayInput{
+				InternetGatewayId: igw.InternetGateways[0].InternetGatewayId,
+			},
+		)
+		if err != nil {
+			return errors.WithStackTrace(err)
+		}
+	}
```

Comment:
What if there is more than one? We should probably add a guard here to log some warning and the fact that we are skipping it.

Reply:
There can only be 0 or 1. I will add a statement about skipping if there are 0.